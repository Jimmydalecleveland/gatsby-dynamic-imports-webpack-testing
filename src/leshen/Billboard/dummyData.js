import lifestyle from '../../images/split-content-lifestyle.jpg'
import lifestyleFull from '../../images/billboard-lifestyle-full.jpg'

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z',
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: '(max-width: 706px) 100vw, 706px',
    presentationWidth: 2560,
  },
}

export const FLUID_IMAGE_DATA_FULL = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAGABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMC/8QAFQEBAQAAAAAAAAAAAAAAAAAAAAH/2gAMAwEAAhADEAAAAdULJA//xAAZEAACAwEAAAAAAAAAAAAAAAABAgADBBL/2gAIAQEAAQUCTYTBa1kfXZ3/AP/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABURAQEAAAAAAAAAAAAAAAAAABAR/9oACAECAQE/AYf/xAAcEAACAQUBAAAAAAAAAAAAAAAAEQECEiExQXH/2gAIAQEABj8C0i6ZxTwlJeH/xAAZEAADAQEBAAAAAAAAAAAAAAABESEAQVH/2gAIAQEAAT8hN23nNcLWQCawQJjHn//aAAwDAQACAAMAAAAQBB//xAAWEQEBAQAAAAAAAAAAAAAAAAABEBH/2gAIAQMBAT8QXJ//xAAWEQEBAQAAAAAAAAAAAAAAAAABABH/2gAIAQIBAT8QA2X/xAAaEAEAAgMBAAAAAAAAAAAAAAABACERMWGR/9oACAEBAAE/EGQY0bPrcwQjrko72algtuO3P//Z',
    src: lifestyleFull,
    srcSet: lifestyleFull,
    aspectRatio: 3.116883116883117,
    sizes: '(max-width: 800px) 100vw, 800px',
    presentationWidth: 2560,
  },
}
