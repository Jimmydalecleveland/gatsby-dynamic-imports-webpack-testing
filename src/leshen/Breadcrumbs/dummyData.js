export const CRUMB_DATA = [
  {
    shortName: 'Home',
    path: '/',
  },
  {
    shortName: 'Internet',
    path: '/internet',
  },
  {
    shortName: 'Gigabit',
    path: '/gigabit',
  },
  {
    shortName: 'Gigabit Plans',
    path: '/gigabit-plans',
  },
]
