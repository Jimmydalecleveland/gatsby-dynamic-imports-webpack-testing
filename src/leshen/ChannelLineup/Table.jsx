import React, { memo } from 'react'
import PropTypes from 'prop-types'
import { Lens as Disc } from 'emotion-icons/material'

import * as Styled from './ChannelLineup.styles'
import Link from '../Link'

const Table = memo(({ packages, channels }) => (
  <Styled.Table>
    <Styled.THead>
      <tr>
        <Styled.THLabel>Channels</Styled.THLabel>
        {packages.map((packageItem) => (
          <Styled.THTopSticky key={packageItem.node.name}>
            {packageItem.node.name}
          </Styled.THTopSticky>
        ))}
      </tr>
    </Styled.THead>
    <tbody>
      {channels.map(({ node: channel }) => (
        <tr key={channel.name}>
          <Styled.THSideSticky>
            {channel.url ? (
              <Link to={channel.url}>{channel.name}</Link>
            ) : (
              channel.name
            )}
          </Styled.THSideSticky>
          {packages.map(({ node: packageItem }) => (
            <Styled.TD key={packageItem.brandy_id}>
              {channel.packageAvailability.includes(packageItem.brandy_id) && (
                <Disc size="0.75em" />
              )}
            </Styled.TD>
          ))}
        </tr>
      ))}
    </tbody>
  </Styled.Table>
))

Table.propTypes = {
  channels: PropTypes.arrayOf(
    PropTypes.shape({
      node: PropTypes.shape({
        name: PropTypes.string.isRequired,
        brandy_id: PropTypes.string.isRequired,
        packageAvailability: PropTypes.arrayOf(PropTypes.string).isRequired,
      }),
    })
  ).isRequired,
  packages: PropTypes.arrayOf(
    PropTypes.shape({
      node: PropTypes.shape({
        name: PropTypes.string.isRequired,
        brandy_id: PropTypes.string.isRequired,
      }),
    })
  ).isRequired,
}

export default Table
