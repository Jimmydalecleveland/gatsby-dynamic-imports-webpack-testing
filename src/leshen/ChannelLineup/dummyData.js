const CHANNEL_LINEUP_DATA = {
  channels: {
    edges: [
      {
        node: {
          name: 'History Channel',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'history-channel',
        },
      },
      {
        node: {
          name: 'Nickelodeon ',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'FX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fx',
        },
      },
      {
        node: {
          name: 'ESPN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'espn',
        },
      },
      {
        node: {
          name: 'FOX',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'fox',
        },
      },
      {
        node: {
          name: 'CNN',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cnn',
        },
      },
      {
        node: {
          name: 'Cartoon Network',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'cartoon-network',
        },
      },
      {
        node: {
          name: 'Comedy Central',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'comedy-central',
          url: '/tv/channels/comedy-central',
        },
      },
      {
        node: {
          name: 'Nickelodeon / Nick At Nite - West (NICKW)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'nickelodeon',
        },
      },
      {
        node: {
          name: 'Turner Network Television (TNT)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'tnt',
          url: '/tv/channels/tnt',
        },
      },
      {
        node: {
          name: 'TV Land (TVLND)',
          packageAvailability: ['dn-americastop120+', 'dn-americastop250'],
          brandy_id: 'tvland',
        },
      },
      {
        node: {
          name: 'E! Entertainment Television (E!)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
          ],
          brandy_id: 'e',
        },
      },
      {
        node: {
          name: 'Syfy Channel (SYFY)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'syfy',
        },
      },
      {
        node: {
          name: 'Eternal Word Television Network (EWTN)',
          packageAvailability: [
            'dn-americastop120',
            'dn-americastop120+',
            'dn-americastop200',
            'dn-americastop250',
          ],
          brandy_id: 'ewtn',
        },
      },
    ],
  },
  packages: {
    edges: [
      {
        node: {
          name: "America's Top 120",
          brandy_id: 'dn-americastop120',
        },
      },
      {
        node: {
          name: "America's Top 120+",
          brandy_id: 'dn-americastop120+',
        },
      },
      {
        node: {
          name: "America's Top 200",
          brandy_id: 'dn-americastop200',
        },
      },
      {
        node: {
          name: "America's Top 250",
          brandy_id: 'dn-americastop250',
        },
      },
    ],
  },
}

export default CHANNEL_LINEUP_DATA
