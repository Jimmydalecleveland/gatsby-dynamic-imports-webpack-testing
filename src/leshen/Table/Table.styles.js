import styled from '@emotion/styled'

export const Table = styled.table`
  font-size: ${({ theme }) => theme.fonts.body.md.size}px;
  font-weight: ${({ theme }) => theme.fonts.body.md.weight};

  font-family: inherit;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;

  thead tr th {
    border-bottom: 2px solid ${({ theme }) => theme.colors.secondary.base.hex};
    text-align: left;
    font-weight: inherit;
  }

  th,
  td {
    padding: ${({ theme }) => theme.spacing.sm};
    min-width: 125px;
    background-color: white;
    color: ${({ theme }) => theme.colors.dark.base.hex};
    border: 0;
  }

  td {
    border-bottom: 1px solid ${({ theme }) => theme.colors.light.base.hex};
    white-space: pre-wrap;
  }

  td.sticky,
  th.sticky {
    position: sticky;
    left: 0px;
    min-width: 125px;
    box-shadow: ${({ theme }) => theme.shadow};

    @media (min-width: ${({ theme }) => theme.breakpoints.sm}) {
      box-shadow: none;
    }
  }

  .sticky p {
    max-width: 100px;
  }
`

export const TableWrapper = styled.div`
  border: 2px solid ${({ theme }) => theme.colors.light.base.hex};
  width: 100%;
  position: relative;
  overflow: auto;
  white-space: nowrap;
`
