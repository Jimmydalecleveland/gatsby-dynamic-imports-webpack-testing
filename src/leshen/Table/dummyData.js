export const tableData = {
  headings: [
    'Average Monthly Use',
    'Average rate (cents/kWh from EFL)',
    'Total bill monthly usage',
  ],
  rows: [
    ['500kWh', '12.1', '$121'],
    ['2,000 kWh', '11.9', '$238'],
  ],
}

export const stickyTableData = {
  headings: ['Activity', 'Average File Size', 'Download Time'],
  rows: [
    ['Stream a 4 minute song', '5 MB', '0.4 sec'],
    ['Watch a 5 minute video', '62 MB', '4.96 sec'],
    ['Scroll social media for 1 hour', '156 MB', '12.48 sec'],
    ['Stream a 45 minute episode', '750 MB', '1 minute'],
    ['Watch a 2 hour HD movie', '2GB', '2.6 minutes'],
  ],
  sticky: true,
}
