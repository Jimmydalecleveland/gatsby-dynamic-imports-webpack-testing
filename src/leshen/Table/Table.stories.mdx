import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Section from '../Section'
import Table, { ticketID } from '.'
import { tableData, stickyTableData } from './dummyData'
import ComponentProgress from '../ComponentProgress'
import Typography from '../Typography'
import TableSectionImage from '../../images/contentful-stories/table-section.png'
import TableModuleImage from '../../images/contentful-stories/table-module.png'
import TableFieldsImage from '../../images/contentful-stories/table-fields.png'


<Meta title='Components|Table' component={Table} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# Table
<Preview>
  <Story name="default">
    <Section>
      <Table json={tableData} />
    </Section>
  </Story>
</Preview>

## Props
<Props of={Table} />

## Sticky Table

<Preview>
  <Story name="sticky">
    <Section>
      <p>
        Resize to a mobile viewport to see the sticky/scrolling functionality.
      </p>
      <Table json={stickyTableData} />
    </Section>
  </Story>
</Preview>

## Contentful

<Preview>
  <Story name="contentful">
    <>
      <Typography>In the "Modules" entry create a new entry and select "Basic Table" from the dropdown.</Typography>
      <img src={TableModuleImage} alt="Table module in contentful" />
      <Typography>Now you can name the table and add the <code>JSON</code> that will render as a table on the front end.</Typography>
      <img src={TableFieldsImage} alt="Example of a Table entryss fields" />
    </>
  </Story>
</Preview>
