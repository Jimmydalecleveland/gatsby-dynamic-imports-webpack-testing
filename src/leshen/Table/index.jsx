/* eslint-disable react/no-danger */
import React from 'react'
import PropTypes from 'prop-types'
import uuid from 'uuid/v4'

import * as Styled from './Table.styles'

const addStickyClass = (i, sticky) => (i === 0 && sticky ? 'sticky' : '')

const Table = ({ json, ...props }) => (
  <Styled.TableWrapper>
    <Styled.Table {...props}>
      <thead>
        <tr>
          {json.headings.map((heading, i) => (
            <th key={heading} className={addStickyClass(i, json.sticky)}>
              {heading}
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {json.rows.map((row) => (
          <tr key={uuid()}>
            {row.map((item, i) => (
              <td
                key={uuid()}
                className={addStickyClass(i, json.sticky)}
                dangerouslySetInnerHTML={{ __html: item }}
              />
            ))}
          </tr>
        ))}
      </tbody>
    </Styled.Table>
  </Styled.TableWrapper>
)

Table.propTypes = {
  json: PropTypes.shape({
    headings: PropTypes.arrayOf(PropTypes.string),
    rows: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)),
    sticky: PropTypes.bool,
  }).isRequired,
}

export default Table

export const ticketID = 'CMS-177'
