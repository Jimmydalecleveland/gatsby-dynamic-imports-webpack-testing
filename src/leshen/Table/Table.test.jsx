import React from 'react'
import { render } from 'setup-tests'
import Table from '.'

import { tableData, stickyTableData } from './dummyData'

describe('Table', () => {
  test('Table should render', () => {
    const { container } = render(<Table json={tableData} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  test('Table should render with sticky header', () => {
    const { container } = render(<Table json={stickyTableData} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
