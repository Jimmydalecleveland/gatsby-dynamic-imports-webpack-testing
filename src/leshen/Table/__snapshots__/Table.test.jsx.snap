// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Table should render 1`] = `
.emotion-2 {
  border: 2px solid #f4f5f6;
  width: 100%;
  position: relative;
  overflow: auto;
  white-space: nowrap;
}

.emotion-0 {
  font-size: 16px;
  font-family: inherit;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
}

.emotion-0 thead tr th {
  border-bottom: 2px solid #8036ff;
  text-align: left;
  font-weight: inherit;
}

.emotion-0 th,
.emotion-0 td {
  padding: 16px;
  min-width: 125px;
  background-color: white;
  color: #383d43;
  border: 0;
}

.emotion-0 td {
  border-bottom: 1px solid #f4f5f6;
  white-space: pre-wrap;
}

.emotion-0 td.sticky,
.emotion-0 th.sticky {
  position: -webkit-sticky;
  position: sticky;
  left: 0px;
  min-width: 125px;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
}

@media (min-width:540px) {
  .emotion-0 td.sticky,
  .emotion-0 th.sticky {
    box-shadow: none;
  }
}

.emotion-0 .sticky p {
  max-width: 100px;
}

<div
  class="emotion-2 emotion-3"
>
  <table
    class="emotion-0 emotion-1"
  >
    <thead>
      <tr>
        <th
          class=""
        >
          Average Monthly Use
        </th>
        <th
          class=""
        >
          Average rate (cents/kWh from EFL)
        </th>
        <th
          class=""
        >
          Total bill monthly usage
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class=""
        >
          500kWh
        </td>
        <td
          class=""
        >
          12.1
        </td>
        <td
          class=""
        >
          $121
        </td>
      </tr>
      <tr>
        <td
          class=""
        >
          2,000 kWh
        </td>
        <td
          class=""
        >
          11.9
        </td>
        <td
          class=""
        >
          $238
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Table Table should render with sticky header 1`] = `
.emotion-2 {
  border: 2px solid #f4f5f6;
  width: 100%;
  position: relative;
  overflow: auto;
  white-space: nowrap;
}

.emotion-0 {
  font-size: 16px;
  font-family: inherit;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
}

.emotion-0 thead tr th {
  border-bottom: 2px solid #8036ff;
  text-align: left;
  font-weight: inherit;
}

.emotion-0 th,
.emotion-0 td {
  padding: 16px;
  min-width: 125px;
  background-color: white;
  color: #383d43;
  border: 0;
}

.emotion-0 td {
  border-bottom: 1px solid #f4f5f6;
  white-space: pre-wrap;
}

.emotion-0 td.sticky,
.emotion-0 th.sticky {
  position: -webkit-sticky;
  position: sticky;
  left: 0px;
  min-width: 125px;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
}

@media (min-width:540px) {
  .emotion-0 td.sticky,
  .emotion-0 th.sticky {
    box-shadow: none;
  }
}

.emotion-0 .sticky p {
  max-width: 100px;
}

<div
  class="emotion-2 emotion-3"
>
  <table
    class="emotion-0 emotion-1"
  >
    <thead>
      <tr>
        <th
          class="sticky"
        >
          Activity
        </th>
        <th
          class=""
        >
          Average File Size
        </th>
        <th
          class=""
        >
          Download Time
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="sticky"
        >
          Stream a 4 minute song
        </td>
        <td
          class=""
        >
          5 MB
        </td>
        <td
          class=""
        >
          0.4 sec
        </td>
      </tr>
      <tr>
        <td
          class="sticky"
        >
          Watch a 5 minute video
        </td>
        <td
          class=""
        >
          62 MB
        </td>
        <td
          class=""
        >
          4.96 sec
        </td>
      </tr>
      <tr>
        <td
          class="sticky"
        >
          Scroll social media for 1 hour
        </td>
        <td
          class=""
        >
          156 MB
        </td>
        <td
          class=""
        >
          12.48 sec
        </td>
      </tr>
      <tr>
        <td
          class="sticky"
        >
          Stream a 45 minute episode
        </td>
        <td
          class=""
        >
          750 MB
        </td>
        <td
          class=""
        >
          1 minute
        </td>
      </tr>
      <tr>
        <td
          class="sticky"
        >
          Watch a 2 hour HD movie
        </td>
        <td
          class=""
        >
          2GB
        </td>
        <td
          class=""
        >
          2.6 minutes
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
